pkg/apprepo/apprepos_handler.go

```diff
@@ -227,8 +228,26 @@ func (a *AppRepositoriesHandler) DeleteAppRepository(req *http.Request, repoName
 	if err != nil {
 		return err
 	}
+	appRepo, err := clientset.KubeappsV1alpha1().AppRepositories(repoNamespace).Get(repoName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	hasCredentials := appRepo.Spec.Auth.Header != nil || appRepo.Spec.Auth.CustomCA != nil
+	var propagationPolicy metav1.DeletionPropagation = "Foreground"
+	err = clientset.KubeappsV1alpha1().AppRepositories(repoNamespace).Delete(repoName, &metav1.DeleteOptions{
+		PropagationPolicy: &propagationPolicy,
+	})
+	if err != nil {
+		return err
+	}
 
-	return clientset.KubeappsV1alpha1().AppRepositories(repoNamespace).Delete(repoName, &metav1.DeleteOptions{})
+	// If the app repo was in a namespace other than the kubeapps one, we also delete the copy of
+	// the repository credentials kept in the kubeapps namespace (the repo credentials in the actual
+	// namespace should be deleted when the owning app repo is deleted).
+	if hasCredentials && repoNamespace != a.kubeappsNamespace {
+		err = clientset.CoreV1().Secrets(a.kubeappsNamespace).Delete(kubeappsSecretNameForRepo(repoName, repoNamespace), &metav1.DeleteOptions{})
+	}
+	return err
```

Comment:
what I have done other times is just try to delete the secret and ignore the error if it's a NotFound. That way you skip one request and in case we create the secret because any other reason (apart than the Auth.Header or the CustomCA) we are sure we delete it.